src/php_spx.h

```diff
@@ -20,7 +20,13 @@
 #   include "config.h"
 #endif
 
+#ifdef _WIN32
+#	pragma warning(push, 0)
+#endif
 #include "main/php.h"
+#ifdef _WIN32
+#	pragma warning(pop)
+#endif
```

Comment:
Including (main/)php.h is not supposed to trigger any serious warnings; what happens in your case?

Reply:
See here https://github.com/NoiseByNorthwest/php-spx/actions/runs/10761275107/job/29840278337

For instance
```
C:\php-src\Zend\zend_config.w32.h(23): error C2220: the following warning is treated as an error (compiling source file ext\php-spx\src/spx_profiler.c)
C:\php-src\Zend\zend_config.w32.h(23): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_profiler.c)
C:\php-src\Zend\zend_portability.h(47): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_config.c)
C:\php-src\Zend\zend_portability.h(47): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_php.c)
C:\php-src\Zend\zend_portability.h(47): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_profiler.c)
C:\php-src\Zend\zend_portability.h(47): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_metric.c)
C:\php-src\Zend\zend_types.h(405): warning C4201: nonstandard extension used: nameless struct/union (compiling source file ext\php-spx\src/spx_profiler.c)
C:\php-src\Zend\zend_types.h(405): warning C4201: nonstandard extension used: nameless struct/union (compiling source file ext\php-spx\src/spx_metric.c)
C:\php-src\Zend\zend_types.h(405): warning C4201: nonstandard extension used: nameless struct/union (compiling source file ext\php-spx\src/spx_config.c)
C:\php-src\Zend\zend_types.h(405): warning C4201: nonstandard extension used: nameless struct/union (compiling source file ext\php-spx\src/spx_php.c)
C:\php-src\Zend\zend_alloc.h(27): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_metric.c)
C:\php-src\Zend\zend_alloc.h(27): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_config.c)
C:\php-src\Zend\zend_alloc.h(27): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_php.c)
C:\php-src\Zend\zend_alloc.h(27): warning C4464: relative include path contains '..' (compiling source file ext\php-spx\src/spx_profiler.c)
C:\php-src\Zend\zend_gc.h(156): warning C4242: '=': conversion from '__int64' to 'int', possible loss of data (compiling source file ext\php-spx\src/spx_config.c)
C:\php-src\Zend\zend_gc.h(156): warning C4242: '=': conversion from '__int64' to 'int', possible loss of data (compiling source file ext\php-spx\src/spx_metric.c)
C:\php-src\Zend\zend_gc.h(156): warning C4242: '=': conversion from '__int64' to 'int', possible loss of data (compiling source file ext\php-spx\src/spx_php.c)
C:\php-src\Zend\zend_gc.h(156): warning C4242: '=': conversion from '__int64' to 'int', possible loss of data (compiling source file ext\php-spx\src/spx_profiler.c)
C:\php-src\Zend\zend_hash.h(15[98](https://github.com/NoiseByNorthwest/php-spx/actions/runs/10761275107/job/29840278337#step:9:99)): warning C4127: conditional expression is constant (compiling source file ext\php-spx\src/spx_config.c)
C:\php-src\Zend\zend_hash.h(1598): warning C4127: conditional expression is constant (compiling source file ext\php-spx\src/spx_php.c)
C:\php-src\Zend\zend_hash.h(1598): warning C4127: conditional expression is constant (compiling source file ext\php-spx\src/spx_metric.c)
C:\php-src\Zend\zend_hash.h(1598): warning C4127: conditional expression is constant (compiling source file ext\php-spx\src/spx_profiler.c)
C:\php-src\Zend\zend_ast.h(374): warning C4242: 'function': conversion from 'uint32_t' to 'zend_ast_attr', possible loss of data (compiling source file ext\php-spx\src/spx_profiler.c)
C:\php-src\Zend\zend_ast.h(374): warning C4242: 'function': conversion from 'uint32_t' to 'zend_ast_attr', possible loss of data (compiling source file ext\php-spx\src/spx_config.c)
```

Reply:
These are from a quick glance, relatively harmless (level 3 and level 4). If you want to keep the `/WX` flag, consider to add `/W2` (that will disable all level 3 and level 4 warnings). Default for more recent PHP versions is `/WX /W1`.

Reply:
OK :+1: